go/mysql/collations/env.go

```diff
@@ -246,3 +220,40 @@ func Local() *Environment {
 	})
 	return defaultEnv
 }
+
+// DefaultConnectionCharset is the default 1-byte charset flag that will be sent in all
+// MySQL clients created by Vitess. This is hardcoded to `utf8mb4_0900_ai_ci` because
+// it's the default in MySQL 8.0+. Older versions of MySQL that do not support this
+// charset will automatically fall back to their built-in default.
+const DefaultConnectionCharset = CollationUtf8mb4ID
+
+// A few interesting character set values.
+// See http://dev.mysql.com/doc/internals/en/character-set.html#packet-Protocol::CharacterSet
+const (
+	CollationUtf8ID    = 33
+	CollationUtf8mb4ID = 255
+	CollationBinaryID  = 63
+)
+
+func ParseConnectionCharset(csname string) (uint8, error) {
+	csname = strings.ToLower(csname)
+	switch csname {
+	case "utf8mb4":
+		return CollationUtf8mb4ID, nil
+	case "utf8":
+		return CollationUtf8ID, nil
+	case "latin1":
+		return 8, nil
+	case "":
+		return DefaultConnectionCharset, nil
+	default:
+		for id, meta := range globalVersionInfo {
+			for _, alias := range meta.alias {
+				if alias == csname && id < 256 {
+					return uint8(id), nil
+				}
+			}
+		}
+		return 0, fmt.Errorf("unsupported connection charset: %q", csname)
+	}
```

Comment:
If I understand correctly, Vitess will not support charsets with an ID > 256, for instance, `utf8mb4_es_0900_ai_ci` will not be usable by someone who explicitly sets the `collation` flags on both VTGate and VTTablet.
The fact that we are removing the `set collation_connection` post-handshake kind of bugs me. 
I understand we cannot use such ID for the handshake since we only have 1 byte available. However, to support collations with an ID greater than 256 could we reestablish the `set collation_connection`, or are we dropping support for them?